nats-base-client/jsclient.ts

```diff
@@ -751,9 +763,25 @@ export class JetStreamSubscriptionImpl extends TypedSubscription<JsMsg>
         `${Js409Errors.IdleHeartbeatMissed}: ${v}`,
         this.sub.subject,
       );
-      this.sub.callback(null, msg);
-      // if we are a handler, we'll continue reporting
-      // iterators will stop
+      const ordered = this.info?.ordered;
+      // non-ordered consumers are always notified of the condition
+      // as they need to try and recover
+      if (!ordered) {
+        this.sub.callback(null, msg);
+      } else {
+        if (!this.js.nc.protocol.connected) {
+          // we are not connected don't do anything
+          return false;
+        }
+        // reset the consumer
+        const seq = this.info?.ordered_consumer_sequence?.delivery_seq;
```

Comment:
this is possibly not right either - the sequence if we have one, should be incremeted